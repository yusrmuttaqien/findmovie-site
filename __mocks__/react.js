import { vi } from 'vitest';

export * from 'react';
export const useEffect = vi.fn((fn) => fn());
export const useRef = vi.fn(() => ({
  current: {},
}));
export const useState = vi.fn((v) => [v, vi.fn()]);
