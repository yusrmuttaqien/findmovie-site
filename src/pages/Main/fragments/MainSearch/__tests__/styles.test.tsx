import { describe, test, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import {
  Wrapper,
  Group,
  Loading,
  Pagination,
  Avatar,
} from 'pages/Main/fragments/MainSearch/styles';
import theme from 'styles/index';
import 'jest-styled-components';

vi.mock('components/Loading', () => ({ default: vi.fn(() => <div />) }));
vi.mock('components/Pagination', () => ({ default: vi.fn(() => <div />) }));
vi.mock('components/Avatar', () => ({ default: vi.fn(() => <figure />) }));

describe('Main page fragment: MainSearch styles', () => {
  test('styled components should match snapshot', () => {
    const { asFragment: fWrapper } = render(<Wrapper theme={theme} />);
    const { asFragment: fGroup } = render(<Group theme={theme} />);
    const { asFragment: fLoading } = render(<Loading theme={theme} />);
    // @ts-expect-error
    const { asFragment: fPagination } = render(<Pagination theme={theme} />);
    // @ts-expect-error
    const { asFragment: fAvatar } = render(<Avatar theme={theme} />);

    expect(fWrapper()).toMatchSnapshot();
    expect(fGroup()).toMatchSnapshot();
    expect(fLoading()).toMatchSnapshot();
    expect(fPagination()).toMatchSnapshot();
    expect(fAvatar()).toMatchSnapshot();
  });
});
