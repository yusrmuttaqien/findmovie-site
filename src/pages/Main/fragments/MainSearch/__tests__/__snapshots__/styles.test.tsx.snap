// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main page fragment: MainSearch styles > styled components should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  padding: 0 1.875rem;
}

@media screen and (min-width: 1279px) {
  .c0 {
    padding: 0 5.5rem;
  }
}

@media screen and (min-width: 1279px) {

}

@media screen and (min-width: 1279px) {

}

<section
    class="c0"
  />
</DocumentFragment>
`;

exports[`Main page fragment: MainSearch styles > styled components should match snapshot 2`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: column;
  gap: 0.9375rem;
}

.c0 #search-title {
  display: flex;
  align-items: baseline;
  flex-wrap: wrap;
  row-gap: 0.1rem;
}

.c0 #search-title p {
  font-size: 0.85rem;
  font-weight: 200;
}

.c0 #search-title h3 {
  font-weight: 500;
  font-size: 1rem;
}

.c0 #categories-wrapper {
  display: inherit;
  flex-direction: inherit;
  gap: 0.625rem;
}

.c0 #categories-wrapper #categories {
  display: inherit;
  flex-direction: inherit;
  position: relative;
  gap: inherit;
}

.c0 #categories-wrapper #categories[data-categories='people'] >div {
  flex-direction: row;
  flex-wrap: wrap;
}

.c0 #categories-wrapper #categories h3 {
  font-weight: 500;
  font-size: 1rem;
  position: sticky;
  top: 0;
  left: 0;
  z-index: 2;
  backdrop-filter: blur(5px);
  background-color: rgba(0, 0, 0, 0.8);
}

.c0 #categories-wrapper #categories >div {
  display: inherit;
  flex-direction: column;
  gap: 0.625rem;
  z-index: 1;
}

@media screen and (min-width: 1279px) {

}

@media screen and (min-width: 1279px) {
  .c0 {
    gap: 2rem;
  }

  .c0 #search-title p {
    font-size: 2rem;
  }

  .c0 #search-title h3 {
    font-size: 2.125rem;
  }

  .c0 #categories-wrapper {
    gap: 1.25rem;
  }

  .c0 #categories-wrapper #categories[data-categories='people'] >div {
    display: flex;
  }

  .c0 #categories-wrapper #categories h3 {
    font-size: 2.125rem;
  }

  .c0 #categories-wrapper #categories >div {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 1.25rem;
  }
}

@media screen and (min-width: 1279px) {

}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`Main page fragment: MainSearch styles > styled components should match snapshot 3`] = `
<DocumentFragment>
  @media screen and (min-width: 1279px) {

}

@media screen and (min-width: 1279px) {

}

@media screen and (min-width: 1279px) {

}

<div />
</DocumentFragment>
`;

exports[`Main page fragment: MainSearch styles > styled components should match snapshot 4`] = `
<DocumentFragment>
  @media screen and (min-width: 1279px) {

}

@media screen and (min-width: 1279px) {

}

@media screen and (min-width: 1279px) {

}

<div />
</DocumentFragment>
`;

exports[`Main page fragment: MainSearch styles > styled components should match snapshot 5`] = `
<DocumentFragment>
  @media screen and (min-width: 1279px) {

}

@media screen and (min-width: 1279px) {

}

@media screen and (min-width: 1279px) {

}

<figure />
</DocumentFragment>
`;
